src/nubank/workspaces/shadow_cljs/target.clj

```diff
@@ -0,0 +1,51 @@
+(ns nubank.workspaces.shadow-cljs.target
+  (:refer-clojure :exclude (compile flush resolve))
+  (:require
+    [shadow.build :as build]
+    [shadow.build.modules :as modules]
+    [shadow.build.targets.browser :as browser]
+    [shadow.build.test-util :as tu]))
+
+;; Mostly taken from shadow-cljs :browser-test target type.
+(defn modify-config [{::build/keys [config] :as state}]
+  (-> state
+    (assoc-in [::build/config :modules :main] {:entries '[nubank.workspaces.shadow-cljs.mount]})
+    (assoc-in [::build/config :compiler-options :source-map] true) ;; always
+    (assoc-in [::build/config :compiler-options :static-fns] false) ;; for mocking
+    (update :build-options merge {:greedy          true
+                                  :dynamic-resolve true})
+    (update-in [::build/config :devtools] merge
+      {:before-load 'nubank.workspaces.core/before-load
+       :after-load  'nubank.workspaces.core/after-load})))
+
+(defn resolve-cards-and-tests
+  [{::build/keys [mode config] :as state}]
+  (let [{:keys [ns-regexp] :or {ns-regexp "-(ws|test)$"}}
+        config
+
+        dynamically-resolved-namespaces
+        (tu/find-namespaces-by-regexp state ns-regexp)]
+
+    (-> state
+      ;; Add the mounter and all of the resolved cards/tests
+      (update-in [::modules/config :main :entries] into dynamically-resolved-namespaces)
```

Comment:
Using `update-in/into` is a bit problematic here since this is called on every compile cycle and you'll keep adding the same namespaces over and over again. They will be de-duped but only after doing extra work and also can never delete a namespace that way once it was added.

Try this instead:
```
(assoc-in [::modules/config :main :entries]
  (-> '[]
      (into dynamically-resolved-namespaces)
      (conj 'nubank.workspaces.shadow-cljs.mount)))
```

Reply:
cool, thanks, I'm going to start using this today on the project, I'll try with that before cutting a release